auth/downscoped/token_broker.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package downscopedoverview
+
+// [START auth_downscoping_token_broker]
+
+import (
+	"context"
+	"fmt"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/google/downscope"
+)
+
+// createDownscopedToken would be run on the token broker in order to generate
+// a downscoped access token.  The token broker would then pass the newly created
+// token to the requesting token consumer for use.
+func createDownscopedToken(bucketName string) error {
+	ctx := context.Background()
+	// A condition can optionally be provided to further restrict access permissions.
+	condition := downscope.AvailabilityCondition{
+		Expression:  "resource.name.startsWith('projects/_/buckets/foo/objects/profile-picture-'",
```

Comment:
Use the bucket name passed into this function in the expression. Also do not hard code the object prefix.